compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -1541,13 +1536,26 @@ impl EnumMemberDescriptionFactory<'ll, 'tcx> {
                 ..
             } => {
                 let tag_info = if fallback {
-                    RegularTag {
+                    // For MSVC, we generate a union of structs for each variant with an explicit
+                    // discriminant field roughly equivalent to the following C:
+                    // ```c
+                    // union enum$<{name}> {
+                    //   struct {variant 0 name} {
+                    //     tag$ variant$;
```

Comment:
I still think it would be more consistent if the `variant$` field here and the `discriminant` field in the niche-layout union would use the same name (e.g. `discriminant$` or `tag$`) 😃

Reply:
Eventually, I would like to have all enums use the `[variant]` synthetic item to name the variant they are currently in and hide all of the `discriminant` fields. This is how it currently works for niche-layout enums but not for directly tagged ones. This will require some more work on the directly tagged case and I'd like to go ahead and get this merged since I don't think it's worth holding this improvement up for that as `variant$` is still better than `RUST$ENUM$DISR` which we have today. 

I would also rather not change the niche-layout ones as `discriminant` is the correct terminology to use in that case. Feel free to r- if you think we should still do something different here!

Reply:
@wesleywiser So are you planning more changes for enums?

Reply:
Yes, I'm hoping to have time this week to get to it.

Reply:
IMO, it'd be nice to move discriminant out of the variant structs.  Otherwise it can be seen when displaying fields of the variant struct, and hiding it is kinda annoying.

Also, I notice that tuple struct variants are not tagged as such, which means that tuple visualizers won't be applied to them.
 (i.e. `enum$<EnumName>::VariantX` should be `tuple<enum$<EnumName>::VariantX>`?)